### PR TITLE
[node] Add global DOMException, ReadableStream and others

### PR DESCRIPTION
**Pull request by ikeyan**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/api/globals.html
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`. no

### CODE REVIEW

File: types/node/globals.d.ts [hunk 1]
```diff
@@ -11,10 +11,79 @@ type _RequestInit = typeof globalThis extends { onmessage: any } ? {}
...
+    readonly ABORT_ERR: 20;
+    readonly URL_MISMATCH_ERR: 21;
+    readonly QUOTA_EXCEEDED_ERR: 22;
+    readonly TIMEOUT_ERR: 23;
+    readonly INVALID_NODE_TYPE_ERR: 24;
+    readonly DATA_CLONE_ERR: 25;
+}
+interface NodeDOMExceptionConstructor {
+    prototype: DOMException;
+    new(message?: string, name?: string): DOMException;
```
Comment (Semigradsky) [inline types/node/globals.d.ts:58]: The second argument can be the object with `cause` and `name` properties.

https://github.com/nodejs/node/pull/44703

Comment (ikeyan) [reply types/node/globals.d.ts]: thanks!

**Review by Semigradsky — APPROVED** [review]

**Review by gabritto — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by ikeyan** [comment]
Ready to merge

**Comment by Renegade334** [comment]
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/6a4c63113c61b088ec4a0d282fe4a982ab810acc/types/node/stream/web.d.ts#L464-L476

@ikeyan, can I just clarify why this one definition needs an infer constraint on DecompressionStream, whereas the equivalents in v20/v18 (and the other interfaces) don't?

**Comment by d43vid** [comment]
> https://github.com/DefinitelyTyped/DefinitelyTyped/blob/6a4c63113c61b088ec4a0d282fe4a982ab810acc/types/node/stream/web.d.ts#L464-L476
> 
> @ikeyan, can I just clarify why this one definition needs an infer constraint on DecompressionStream, whereas the equivalents in v20/v18 (and the other interfaces) don't?

I'm also wondering why that was done. Yesterday our builds have started failing because of this:

node_1  | node_modules/@types/node/stream/web.d.ts:469:56 - error TS1005: '?' expected.
node_1  | 
node_1  | 469             DecompressionStream: infer T extends object;

Possibly this slipped through and shouldn't have been added.

**Comment by Renegade334** [comment]
> Yesterday our builds have started failing because of this:
> 
> node_1  | node_modules/@types/node/stream/web.d.ts:469:56 - error TS1005: '?' expected.
> node_1  | 
> node_1  | 469             DecompressionStream: infer T extends object;

@d43vid: that error suggests that you're using TypeScript 4.6 or prior, which are beyond the two-year support window for DefinitelyTyped packages.

**Comment by d43vid** [comment]
> > Yesterday our builds have started failing because of this:
> > node_1  | node_modules/@types/node/stream/web.d.ts:469:56 - error TS1005: '?' expected.
> > node_1  |
> > node_1  | 469             DecompressionStream: infer T extends object;
> 
> @d43vid: that error suggests that you're using TypeScript 4.6 or prior, which are beyond the two-year support window for DefinitelyTyped packages.

Thank you very much, that was indeed the case. After updating to a new version everything runs smoothly again.

**Comment by ikeyan** [comment]
@Renegade334 I overlooked that difference. no problem deleting the constraint.

**Pull request merged**